packages/react-components/react-tree/stories/A_Tree/TreeMultiSelection.stories.tsx

```diff
@@ -0,0 +1,54 @@
+import * as React from 'react';
```

Comment:
Something odd is happpening here

![nested tree selection](https://github.com/microsoft/fluentui/assets/20744592/0f7d678a-3b03-413d-ae5c-3b2ba1c70823)

Reply:
Great catch, I'll create a test for it.

Reply:
Apparently when I split this PR with this one https://github.com/microsoft/fluentui/pull/28669 I forgot that I need the logic to have the `checked` state of the parent on context to determine if it's children are checked or not.

I'll block this to merge the other PR first